tools/cosmovisor/scanner.go

```diff
@@ -113,18 +113,22 @@ func (fw *fileWatcher) CheckUpdate(currentUpgrade upgradetypes.Plan) bool {
 		return false
 	}
 
+	// check https://github.com/cosmos/cosmos-sdk/issues/21086
+	// If new file is still empty, wait a small amount of time for write to complete
+	if stat.Size() == 0 {
+		time.Sleep(2 * time.Millisecond)
+		stat, err = os.Stat(fw.filename)
+		if err != nil {
+			// file doesn't exists
+			return false
+		}
+	}
```

Comment:
maybe here instead of waiting a fixed amount of time, it could run the check in a loop just to be sure:

```go
attempts := 0
for attempts < maxAttempts {
    stat, err := os.Stat(fw.filename)
    if err != nil {
        if os.IsNotExist(err) {
            return false
        }
        return false
    }

    if stat.Size() > 0 {
        break
    }

    time.Sleep(waitTime)
    attempts++
}

if attempts == maxAttempts {
    // exit
    return false
}
```

Reply:
adding commit now